pennylane/ops/qubit/state_preparation.py

```diff
@@ -88,6 +91,24 @@ def compute_decomposition(n, wires):
         """
         return [BasisStatePreparation(n, wires)]
 
+    def matrix(self, wire_order=None):
+        """Returns a ket vector representing the state being created."""
+        if wire_order is None:
+            num_wires = len(self.wires)
+            indices = self.parameters[0]
+        else:
+            wires = Wires(wire_order)
+            if not wires.contains_wires(self.wires):
+                raise WireError("Custom wire_order must contain all BasisState wires")
+            num_wires = len(wires)
+            indices = [0] * num_wires
+            for wire_idx, value in zip(self.wires, self.parameters[0]):
```

Comment:
Probably worth to rename wires_idx, because it looks confusing to have `indices[wires.index(wire_idx)]`. It would be nice to have a good separation bewteen the self.wires and the wires from the given order.